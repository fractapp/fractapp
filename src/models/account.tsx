import { Keyring } from '@polkadot/api';
import { KeyringPair } from '@polkadot/keyring/types';



/*
const keyring = new Keyring({ type: "sr25519" })

interface CurrencyAdapter {
    address(): string;
}

class PolkadotAndCusamaAdapter implements CurrencyAdapter {
    keyPair: KeyringPair
    constructor(seed: string, currency: Currency) {
        switch (currency) {
            case Currency.Polkadot:
                keyring.setSS58Format(0)
                break;
            case Currency.Kusama:
                keyring.setSS58Format(0)
                break;
        }
        this.keyPair = keyring.addFromUri(seed)
    }

    address() {
        return this.keyPair.address
    }
}

export class Account {
    seed: string;
    adapters: Map<Currency, CurrencyAdapter>

    constructor(seed: string, adapters?: Map<Currency, CurrencyAdapter>) {
        this.seed = seed;
        if (adapters == null) {
            this.adapters = new Map<Currency, CurrencyAdapter>(
                [
                    [Currency.Polkadot, new PolkadotAndCusamaAdapter(seed, Currency.Polkadot)],
                    [Currency.Kusama, new PolkadotAndCusamaAdapter(seed, Currency.Kusama)]
                ]
            )
        } else {
            this.adapters = adapters;
        }
    }


    address(currency: Currency) {
        adapters
    }


    backupAccount(password: String) {

    }
}
*/