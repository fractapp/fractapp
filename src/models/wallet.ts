import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
