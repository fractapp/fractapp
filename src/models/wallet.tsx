import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
