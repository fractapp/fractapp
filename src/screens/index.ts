export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
