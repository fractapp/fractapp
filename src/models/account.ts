import { Currency } from "./wallet";

/**
 * @category Models
 */
export class Account {
    name: string;
    address: string;
    pubKey: string;
    currency: Currency;
    balance: number;

    constructor(name: string, address: string, pubKey: string, currency: Currency, balance: number) {
        this.name = name
        this.address = address
        this.pubKey = pubKey
        this.currency = currency
        this.balance = balance
    }

    static parse(json: string): Account {
        return JSON.parse(json)
    }
}