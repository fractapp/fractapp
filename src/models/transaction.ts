import {Currency} from './wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export class Transaction {
  id: string;
  member: string;
  currency: Currency;
  value: number;
  txType: TxType;
  timestamp: number;
  fee: number;
  usdValue: number;
  usdFee: number;
  status: TxStatus;

  constructor(
    id: string,
    member: string,
    currency: Currency,
    txType: TxType,
    timestamp: number,
    value: number,
    usdValue: number,
    fee: number,
    usdFee: number,
    status: TxStatus,
  ) {
    this.id = id;
    this.member = member;
    this.currency = currency;
    this.value = value;
    this.txType = txType;
    this.timestamp = timestamp;
    this.fee = fee;
    this.usdValue = usdValue;
    this.usdFee = usdFee;
    this.status = status;
  }
}
