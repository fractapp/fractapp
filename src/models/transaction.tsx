import { ImageSourcePropType } from "react-native";
import { Currency } from "./wallet";

export enum TxType {
    None = 0,
    Sent,
    Received
}

export class Transaction {
    id: string;
    member: string;
    currency: Currency;
    value: number;
    txType: TxType;
    date: Date;

    usdValue: number;
    symbol: string;
    logo: ImageSourcePropType;

    constructor(id: string, member: string, currency: Currency, txType: TxType, date: Date, value: number) {
        this.id = id;
        this.member = member;
        this.currency = currency;
        this.value = value;
        this.txType = txType;
        this.date = date;
        //TODO get state
        let rate = 10
        this.usdValue = value * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
