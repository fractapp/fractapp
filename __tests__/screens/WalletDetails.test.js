import React, {useContext, useState} from 'react';
import renderer from 'react-test-renderer';
import {WalletDetails} from 'screens/WalletDetails';
import {Currency, Wallet} from 'types/wallet';
import GlobalStore, {initialState} from 'storage/Global';
import ChatsStore from 'storage/Chats';
import {TxStatus, TxType} from 'types/transaction';
import {fireEvent, render} from '@testing-library/react-native';
import {Network} from 'types/account';

jest.mock('storage/DB', () => ({}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useContext: jest.fn(),
  Linking: jest.fn(() => ({
    openURL: jest.fn(),
  })),
}));
jest.mock('utils/api', () => ({
  getImgUrl: jest.fn(() => 'userAvatarMock'),
}));

useState.mockImplementation((init) => [init, jest.fn()]);

it('Test view with empty state', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
  });
  useContext.mockReturnValueOnce({
    state: ChatsStore.initialState(),
  });

  const tree = renderer
    .create(
      <WalletDetails
        navigation={null}
        route={{
          params: {
            wallet: new Wallet(
              'Wallet Polkadot',
              'address#1',
              Currency.DOT,
              1,
              100,
              '1000000000000',
              10,
            ),
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view with txs', () => {
  const chatsState = ChatsStore.initialState();
  chatsState.transactions.set(Currency.DOT, {
    transactionById: new Map([
      [
        '1',
        {
          id: '1',
          userId: null,
          address: 'address#1',
          currency: Currency.DOT,
          txType: TxType.Sent,
          timestamp: new Date('02-12-2020').getTime(),
          value: 12,
          usdValue: 12,
          fee: 12,
          usdFee: 12,
          status: TxStatus.Fail,
        },
      ],
    ]),
  });
  chatsState.transactions.set(Currency.KSM, {
    transactionById: new Map([
      [
        '2',
        {
          id: '2',
          userId: 'userId',
          address: 'address#2DOT',
          currency: Currency.KSM,
          txType: TxType.Sent,
          timestamp: new Date('03-12-2020').getTime(),
          value: 10,
          usdValue: 10,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Success,
        },
      ],
    ]),
  });
  const globalState = GlobalStore.initialState();
  globalState.users.set('userId', {
    id: 'id',
    name: 'name',
    username: 'username',
    avatarExt: 'png',
    lastUpdate: new Date('12-12-2020').getTime(),
    addresses: {
      0: 'address#2DOT',
      1: 'address#2KSM',
    },
  });
  useContext.mockReturnValueOnce({
    state: globalState,
    dispatch: () => null,
  });
  useContext.mockReturnValueOnce({
    state: {
      transactions: chatsState.transactions,
      isInitialized: true,
    },
    dispatch: () => null,
  });

  const tree = renderer
    .create(
      <WalletDetails
        navigation={null}
        route={{
          params: {
            wallet: new Wallet(
              'Wallet Polkadot',
              'address#1',
              Currency.DOT,
              100,
              '1000000000000',
              10,
            ),
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view click send', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
  });
  useContext.mockReturnValueOnce({
    state: ChatsStore.initialState(),
  });

  const wallet = new Wallet(
    'Wallet Polkadot',
    'address#1',
    Currency.DOT,
    100,
    '1000000000000',
    10,
  );
  const navigate = jest.fn();
  const component = render(
    <WalletDetails
      navigation={{
        navigate: navigate,
      }}
      route={{
        params: {
          wallet: wallet,
        },
      }}
    />,
  );

  fireEvent.press(component.getByTestId('sendBtn'));
  expect(navigate).toBeCalledWith('Search', {
    wallet: wallet,
  });
});

it('Test view click receive', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
  });
  useContext.mockReturnValueOnce({
    state: ChatsStore.initialState(),
  });

  const wallet = new Wallet(
    'Wallet Polkadot',
    'address#1',
    Currency.DOT,
    100,
    '1000000000000',
    10,
  );
  const navigate = jest.fn();
  const component = render(
    <WalletDetails
      navigation={{
        navigate: navigate,
      }}
      route={{
        params: {
          wallet: wallet,
        },
      }}
    />,
  );

  fireEvent.press(component.getByTestId('receiveBtn'));
  expect(navigate).toBeCalledWith('Receive', {
    address: wallet.address,
    currency: wallet.currency,
  });
});

it('Test view click1 txs', () => {
  const wallet = new Wallet(
    'Wallet Polkadot',
    'address#1',
    Currency.DOT,
    Network.Polkadot,
    100,
    '1000000000000',
    10,
  );

  const txs = new Map();
  txs.set(Currency.DOT, {
    id: '1',
    userId: null,
    address: 'address#1',
    currency: Currency.DOT,
    txType: TxType.Sent,
    timestamp: new Date('02-12-2020').getTime(),
    value: 12,
    usdValue: 12,
    fee: 12,
    usdFee: 12,
    status: TxStatus.Fail,
  });

  const chatsState = ChatsStore.initialState();
  chatsState.transactions.set(Currency.DOT, {
    transactionById: new Map([['1', txs.get(Currency.DOT)]]),
  });


  const globalState = GlobalStore.initialState();
  globalState.users.set('userId', {
    id: 'id',
    name: 'name',
    username: 'username',
    avatarExt: 'png',
    lastUpdate: new Date('12-12-2020').getTime(),
    addresses: {
      0: 'address#2DOT',
      1: 'address#2KSM',
    },
  });
  useContext.mockReturnValueOnce({
    state: globalState,
    dispatch: () => null,
  });
  useContext.mockReturnValueOnce({
    state: {
      transactions: chatsState.transactions,
      isInitialized: true,
    },
    dispatch: () => null,
  });

  const navigate = jest.fn();
  const component = render(
    <WalletDetails
      navigation={{
        navigate: navigate,
      }}
      route={{
        params: {
          wallet: wallet,
        },
      }}
    />,
  );

  fireEvent.press(component.getByText('address#1'));
  expect(navigate).toBeCalledWith('TransactionDetails', {
    transaction: txs.get(Currency.DOT),
    wallet: wallet,
    user: null,
  });
});

it('Test view click2 txs', () => {
  const wallet = new Wallet(
    'Wallet Polkadot',
    'address#1',
    Currency.DOT,
    Network.Polkadot,
    100,
    '1000000000000',
    10,
  );

  const txs = new Map();
  txs.set(Currency.DOT, {
    id: '1',
    userId: 'userId',
    address: 'address#1',
    currency: Currency.DOT,
    txType: TxType.Sent,
    timestamp: new Date('03-12-2020').getTime(),
    value: 10,
    usdValue: 10,
    fee: 10,
    usdFee: 10,
    status: TxStatus.Success,
  });

  const chatsState = ChatsStore.initialState();
  chatsState.transactions.set(Currency.DOT, {
    transactionById: new Map([['1', txs.get(Currency.DOT)]]),
  });

  const globalState = GlobalStore.initialState();
  globalState.users.set('userId', {
    id: 'id',
    name: 'name',
    username: 'username',
    avatarExt: 'png',
    lastUpdate: new Date('12-12-2020').getTime(),
    addresses: {
      0: 'address#2DOT',
      1: 'address#2KSM',
    },
  });
  useContext.mockReturnValueOnce({
    state: globalState,
    dispatch: () => null,
  });
  useContext.mockReturnValueOnce({
    state: {
      transactions: chatsState.transactions,
      isInitialized: true,
    },
    dispatch: () => null,
  });

  const navigate = jest.fn();
  const component = render(
    <WalletDetails
      navigation={{
        navigate: navigate,
      }}
      route={{
        params: {
          wallet: wallet,
        },
      }}
    />,
  );
  fireEvent.press(component.getByText('name'));
  expect(navigate).toBeCalledWith('TransactionDetails', {
    transaction: txs.get(Currency.DOT),
    wallet: wallet,
    user: globalState.users.get('userId'),
  });
});
