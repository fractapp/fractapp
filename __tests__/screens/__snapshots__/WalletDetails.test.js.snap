// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view with empty state 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 15,
                "width": "85%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/img/dot.png",
                  }
                }
                style={
                  Object {
                    "borderColor": "black",
                    "borderRadius": 25,
                    "borderWidth": 0.8,
                    "height": 50,
                    "overflow": "hidden",
                    "width": 50,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "marginLeft": 15,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#888888",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    }
                  }
                >
                  Wallet Polkadot
                </Text>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 23,
                            "fontStyle": "normal",
                          },
                          Object {
                            "textAlign": "left",
                          },
                        ]
                      }
                    >
                      100
                       
                      DOT
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 23,
                            "fontStyle": "normal",
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      $
                      1000
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 30,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#F0EEEE",
                    "borderRadius": 45,
                    "height": 70,
                    "justifyContent": "center",
                    "width": 70,
                  }
                }
                testID="sendBtn"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#666666",
                        "fontSize": 24,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰸇
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Send
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginLeft": 30,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F0EEEE",
                      "borderRadius": 45,
                      "height": 70,
                      "justifyContent": "center",
                      "width": 70,
                    },
                  ]
                }
                testID="receiveBtn"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#666666",
                        "fontSize": 24,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰮏
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Receive
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#cccccc",
                "height": 1,
                "marginTop": 25,
                "width": "90%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Test view with txs 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "address": "address#2DOT",
              "currency": 0,
              "fee": 10,
              "id": "2",
              "status": 1,
              "timestamp": 1583960400000,
              "txType": 1,
              "usdFee": 10,
              "usdValue": 10,
              "userId": "userId",
              "value": 10,
            },
          ],
          "title": "12 March 2020",
        },
        Object {
          "data": Array [
            Object {
              "address": "address#1",
              "currency": 0,
              "fee": 12,
              "id": "1",
              "status": 2,
              "timestamp": 1581454800000,
              "txType": 1,
              "usdFee": 12,
              "usdValue": 12,
              "userId": null,
              "value": 12,
            },
          ],
          "title": "12 February 2020",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        1,
        4,
      ]
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 15,
                "width": "85%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/img/dot.png",
                  }
                }
                style={
                  Object {
                    "borderColor": "black",
                    "borderRadius": 25,
                    "borderWidth": 0.8,
                    "height": 50,
                    "overflow": "hidden",
                    "width": 50,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "marginLeft": 15,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#888888",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    }
                  }
                >
                  Wallet Polkadot
                </Text>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 23,
                            "fontStyle": "normal",
                          },
                          Object {
                            "textAlign": "left",
                          },
                        ]
                      }
                    >
                      100
                       
                      DOT
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 23,
                            "fontStyle": "normal",
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      $
                      1000
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 30,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#F0EEEE",
                    "borderRadius": 45,
                    "height": 70,
                    "justifyContent": "center",
                    "width": 70,
                  }
                }
                testID="sendBtn"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#666666",
                        "fontSize": 24,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰸇
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Send
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginLeft": 30,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F0EEEE",
                      "borderRadius": 45,
                      "height": 70,
                      "justifyContent": "center",
                      "width": 70,
                    },
                  ]
                }
                testID="receiveBtn"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#666666",
                        "fontSize": 24,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰮏
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Receive
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#cccccc",
                "height": 1,
                "marginTop": 25,
                "width": "90%",
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Roboto-Bold",
                "fontSize": 17,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "marginBottom": 5,
                "marginLeft": "5%",
              }
            }
          >
            12 March 2020
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingTop": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Image
                height={50}
                source={
                  Object {
                    "uri": "userAvatarMock",
                  }
                }
                style={
                  Object {
                    "borderRadius": 25,
                    "height": 50,
                    "width": 50,
                  }
                }
                width={50}
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    }
                  }
                >
                  name
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "fontStyle": "normal",
                      },
                      Object {
                        "color": "#EA4335",
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  -
                  10
                   
                  DOT
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 15,
                      "fontStyle": "normal",
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#EA4335",
                    },
                  ]
                }
              >
                -
                $
                10
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Roboto-Bold",
                "fontSize": 17,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "marginBottom": 5,
                "marginLeft": "5%",
              }
            }
          >
            12 February 2020
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingTop": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/img/dot.png",
                  }
                }
                style={
                  Object {
                    "borderColor": "black",
                    "borderRadius": 25,
                    "borderWidth": 0.8,
                    "height": 50,
                    "overflow": "hidden",
                    "width": 50,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 23,
                      "bottom": 0,
                      "height": 23,
                      "justifyContent": "center",
                      "left": 30,
                      "position": "absolute",
                      "width": 23,
                    },
                    Object {
                      "borderColor": "#EA4335",
                      "borderWidth": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#EA4335",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅖
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    }
                  }
                >
                  address#1
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "fontStyle": "normal",
                      },
                      Object {
                        "color": "#EA4335",
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  -
                  12
                   
                  DOT
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 15,
                      "fontStyle": "normal",
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#EA4335",
                    },
                  ]
                }
              >
                -
                $
                12
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
